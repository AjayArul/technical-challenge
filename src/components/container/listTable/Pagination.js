import React, { Component } from 'react'
import { func, number, string, array } from 'prop-types'

class Pagination extends Component {

  nextPage = () => {
    let currentPage = this.props.currentPageNum + 1;
    this.props.handlePage(currentPage)
  }

  previousPage = () => { 
    let currentPage = this.props.currentPageNum - 1;
    this.props.handlePage(currentPage)
  };

  render () {
    let { dataList, currentPageNum, dataPerPage, numberOfRecords } = this.props;
    
    // Logic for displaying page numbers
    let totalPages = [];
    dataList && (totalPages = Math.ceil(dataList.length / dataPerPage))

    if (!(dataList.length <= dataPerPage) && !(currentPageNum === 0 || currentPageNum > totalPages)) {
      return (
        <div className="pagination">
          <div className="control left">
            <button id="prevPage" type="button" aria-label="previous page" title="Previous Page" disabled={currentPageNum <= 1 ? true : false} onClick={this.previousPage.bind(this)}>
              <i className="fas fa-chevron-circle-left"></i>
            </button>
          </div>
          <div className="pag-label">
            <span className="first"> {currentPageNum}</span> of <span className="total">{totalPages}</span>
          </div>
          <div className="control right">
            <button id="nexPage" type="button" aria-label="next page" title="Next Page" disabled={currentPageNum == totalPages ? true : false} onClick={this.nextPage.bind(this)}>
              <i className="fas fa-chevron-circle-right"></i>
            </button>
          </div>
        </div >
      );
    } else {
      return null
    }

  }
}

Pagination.propTypes = {
  currentPageNum: number,
  dataPerPage: number,
  dataList: array,
  handlePage: func
};

export default Pagination;
