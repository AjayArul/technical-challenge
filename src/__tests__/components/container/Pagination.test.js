import React from 'react'
import Pagination from './../../../components/container/listTable/Pagination'
import { DATA_LIST } from './../../../Constants'

describe('Pagination wrapper Test <Pagination handlePage="fn()" totalPages=100 currentPageNum=1 dataPerPage=10 />', () => {
  let _props, _wrapper

  beforeEach(() => {
    _props = {
      handlePage: jest.fn(),
      totalPages: 100,
      currentPageNum: 1,
      dataPerPage: 10,
      dataList: DATA_LIST
    }
    _wrapper = shallow(<Pagination {..._props} />)
  })

  it('Should define and render <Pagination /> component', () => {
    expect(_wrapper).toBeDefined()
    expect(_wrapper).toMatchSnapshot()
  })

  it('Should trigger `nextPage` when clicked on next page arrow button and call handlePage with currentPageNum + 1', () => {
    const nextBtn = _wrapper.find('#nexPage')
    nextBtn.simulate('click')
    expect(_props.handlePage).toHaveBeenCalled()
  })

  it('Should trigger `previousPage` when clicked on prevPage arrow button and call handlePage with currentPageNum - 1', () => {
    const prevPage = _wrapper.find('#prevPage')
    prevPage.simulate('click')
    expect(_props.handlePage).toHaveBeenCalled()
  })

  // it('When the `currentPageNum` is more than 1 then the previous button should enable', () => {
  //   _wrapper.setProps({ currentPageNum: 10 })
  //   const prevPage = _wrapper.dive().find('#prevPage')
  //   expect(prevPage.html().includes('disabled')).toBe(false)
  // })

  // it('When the `currentPageNum` and totalPages are equal then next button should disable', () => {
  //   _wrapper.setProps({ currentPageNum: 100 })
  //   const nexPage = _wrapper.find('#nexPage')
  //   expect(nexPage.html().includes('disabled')).toBe(true)
  // })
})
