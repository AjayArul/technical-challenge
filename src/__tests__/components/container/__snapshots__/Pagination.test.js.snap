// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination wrapper Test <Pagination handlePage="fn()" totalPages=100 currentPageNum=1 dataPerPage=10 /> Should define and render <Pagination /> component 1`] = `ShallowWrapper {}`;
